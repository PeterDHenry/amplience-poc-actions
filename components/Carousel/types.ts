import { HeroProps } from '../Hero/types'

export type CarouselProps = {
  heroBannerList: HeroProps[]
}

export const defaultProps: CarouselProps = {
  heroBannerList: [
    {
      _meta: {
        name: 'Default ride like a pro banner',
        schema: 'https://wiggle.com/hero-banner',
        deliveryId: '18a7eee8-6c7a-4a92-8a76-29eb936679a8',
      },
      background: {
        _meta: {
          schema:
            'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
        },
        id: 'ef717f48-99e2-4717-89da-6f906056eae1',
        name: 'CCC-11-5-a',
        endpoint: 'wigglecrcnonprd',
        defaultHost: 'cdn.media.amplience.net',
      },
      textAndCTA: {
        headline: 'Default ride like a pro',
        callToActionURL: 'https://www.wiggle.co.uk/pro-riding-for-summer',
        callToActionText: 'Default shop now',
      },
    },
    {
      _meta: {
        name: 'Default Tailored for women banner',
        schema: 'https://wiggle.com/hero-banner',
        deliveryId: 'ecba6af1-17ce-4799-9d20-146b307af92d',
      },
      background: {
        _meta: {
          schema:
            'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
        },
        id: '9c6bd46b-21e2-4720-8be6-3fb4b9be6079',
        name: 'homepageh1d',
        endpoint: 'wigglecrcnonprd',
        defaultHost: 'cdn.media.amplience.net',
      },
      textAndCTA: {
        headline: 'Default Tailored for women',
        strapline: 'Your best-ever sports collection',
        callToActionText: 'Default Shop now',
        callToActionURL: 'https://www.wiggle.co.uk/womens-collection',
      },
    },
    {
      _meta: {
        name: 'Default Discover different banner',
        schema: 'https://wiggle.com/hero-banner',
        deliveryId: 'a6cac415-3411-4481-a741-c7d4e1edbff2',
      },
      background: {
        _meta: {
          schema:
            'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
        },
        id: 'a11cb660-fd4a-498f-8222-022edfeb479b',
        name: 'homepage-h-d',
        endpoint: 'wigglecrcnonprd',
        defaultHost: 'cdn.media.amplience.net',
      },
      textAndCTA: {
        headline: 'Default Discover different',
        callToActionText: 'Default Find out more',
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
      },
    },
  ],
}
