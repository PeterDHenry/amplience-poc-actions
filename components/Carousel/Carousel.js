import React from 'react';
import styled from 'styled-components'
import Carousel from 'react-bootstrap/Carousel'

import Hero from '../Hero/Hero';

const StyledCarouselItem= styled(Carousel.Item)`
-webkit-transition: -webkit-transform 0s ease-in-out;
-o-transition: -o-transform 0s ease-in-out;
transition: transform 0s ease-in-out;`

export default function Carousels ({ heroBannerList }) {
  
  return ( 
    <div>
      <Carousel controls nextLabel="" prevLabel="" slide controls indicators>
        {(heroBannerList.map((carousel)=>(
          <StyledCarouselItem  key={carousel._meta.deliveryId}>
            <Hero {...carousel}
                  cssClass="carousel item"
            />
          </StyledCarouselItem>
        )))}   
      </Carousel>
    </div>
    )
  };
  
  