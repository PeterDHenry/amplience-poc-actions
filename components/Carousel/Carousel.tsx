import React from 'react'
import styled from 'styled-components'
import bsCarousel from 'react-bootstrap/Carousel'
import { Hero } from '../Hero/Hero'
import { HeroProps } from '../Hero/types'
import { CarouselProps } from './types'

const StyledCarouselItem = styled(bsCarousel.Item)`
  -webkit-transition: -webkit-transform 0s ease-in-out;
  -o-transition: -o-transform 0s ease-in-out;
  transition: transform 0s ease-in-out;
`

const StyledCarouselWithIndicators = styled(bsCarousel)`
  .carousel-indicators li {
    background-color: #fff;
    height: 15px;
    width: 40px;
    margin-left: 10px;
  }

  .carousel-indicators .active {
    background-color: #ff920b;
  }
`

export const Carousel = ({ heroBannerList }: CarouselProps) => {
  return (
    <div>
      <StyledCarouselWithIndicators
        controls
        nextLabel=""
        prevLabel=""
        slide
        indicators
      >
        {heroBannerList.map((hero: HeroProps) => (
          <StyledCarouselItem key={hero._meta.deliveryId}>
            <Hero {...hero} cssClass="carousel item" />
          </StyledCarouselItem>
        ))}
      </StyledCarouselWithIndicators>
    </div>
  )
}
