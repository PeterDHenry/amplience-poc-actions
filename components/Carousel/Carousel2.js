import React, { useState, useEffect } from 'react';
import styled from 'styled-components'
import axios from 'axios';
import { ContentClient } from 'dc-delivery-sdk-js';

import Hero from '../Hero/Hero';

const client = new ContentClient({
  hubName: "sandbox-dev"
});


const SlideHolder = styled.div`
    position: relative;
`;

const SlideInner = styled.div`
    position: relative;
    width: 100%;
    overflow: hidden;
`;

const Slide = styled.div`
    position: relative;
    display: none;
    float: left;
    width: 100%;
    margin-right: -100%;
    -webkit-backface-visibility: hidden;
    backface-visibility: hidden;
    transition: transform .6s ease-in-out;

    &.activated {
        display: block;
    }
`;
   

const Carousel = () => {

  const REQUEST_STATUS = {
    LOADING: 'loading',
    SUCCESS: 'success',
    ERROR: 'error',
  };

  const [status, setStatus] = useState(REQUEST_STATUS.LOADING);
  const [error, setError] = useState({});
  const [carousels, setCarousels] = useState([]);
  const [slot, setSlot] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      try {
        await axios.get('https://sandbox-dev.cdn.content.amplience.net/content/id/2ea7590a-5546-494f-95e9-82910f25276c')
        .then((response) => {
            setSlot(response.data.content);
            return response.data.content;
        })
        .then(async (slot) => {
            for (const slotContent of slot.slotContent) {
                await axios.get(`https://sandbox-dev.cdn.content.amplience.net/content/id/${slotContent.id}`).then((response) => {
                    setCarousels(hero => [...hero, response.data]);
                })
           }
           setStatus(REQUEST_STATUS.SUCCESS);
        })
      } catch (e) {
        setStatus(REQUEST_STATUS.ERROR);
        setError(e);
      }
    };
    fetchData();
  }, []);



  const success = status === REQUEST_STATUS.SUCCESS;
  const isLoading = status === REQUEST_STATUS.LOADING;
  const hasErrored = status === REQUEST_STATUS.ERROR;

  return (
      
    <div>
      {isLoading && <div>Loading...</div>}
      {hasErrored && (
        <div>
          Loading error... Oops something went wrong...
          <br />
          <b>ERROR: {error.message}</b>
        </div>
      )}
      <SlideHolder>
        <SlideInner>
        {success && (carousels.map((carousel, index)=>(
            <Slide className={`${index===0? 'activated':''}`}>
                  <Hero
                      key={carousel.content._meta.deliveryId}
                      {...carousel}
                      cssClass="carousel"
                  />
            </Slide>
          )))
        }
        </SlideInner>      
      </SlideHolder>
    </div>
  )
};

export default Carousel;
