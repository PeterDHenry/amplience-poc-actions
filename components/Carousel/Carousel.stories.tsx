import { Meta, Story } from '@storybook/react'
import { Carousel as Component } from './Carousel'
import { defaultProps, CarouselProps } from './types'

const meta: Meta = {
  title: 'CMS Content',
  component: Component,
  argTypes: {},
}

export default meta

type StoryProps = CarouselProps

export const Carousel: Story<StoryProps> = (props) => <Component {...props} />

Carousel.args = defaultProps
