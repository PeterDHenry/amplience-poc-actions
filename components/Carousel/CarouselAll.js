import React, { useState, useEffect } from 'react';
import styled from 'styled-components'
import axios from 'axios';
import Carousel from 'react-bootstrap/Carousel'

import Hero from '../Hero/Hero';

const StyledCarouselItem= styled(Carousel.Item)`
  -webkit-transition: -webkit-transform 0s ease-in-out;
   -o-transition: -o-transform 0s ease-in-out;
   transition: transform 0s ease-in-out;`

export default function Carousels ({ homeSlot }) {

  const [carousels, setCarousels] = useState([]);

  useEffect(() => {
    for (const slotContent of homeSlot.content.heroBannerList) {
      setCarousels(hero => [...hero, slotContent]);
    }
  }, []);

  return (
      
    <div>
      <Carousel controls nextLabel="" prevLabel="" slide controls indicators>
        {(carousels.map((carousel)=>(
            <StyledCarouselItem  key={carousel._meta.deliveryId}>
              <Hero
                {...carousel}
                cssClass="carousel item"
              />
            </StyledCarouselItem>
            ))
          )
        }   
      </Carousel>
    </div>
  )
};

