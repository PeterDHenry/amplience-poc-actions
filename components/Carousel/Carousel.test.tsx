import { render } from '@testing-library/react'
import Carousel, { CarouselListProps } from './Carousel'
import '@testing-library/jest-dom'

describe('Carousel component', () => {
  test('Matches snapshot', () => {
    const { asFragment } = render(<Carousel {...carouselListProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  test('Check if the Carousel is active and working correctly', () => {
    render(<Carousel {...carouselListProps} />, {})
    const carouselItems =
      document.getElementsByClassName('carousel-item').length
    expect(carouselItems).toBeGreaterThan(0)
    expect(carouselItems).toBeGreaterThan(1)
  })

  test('Check if Carousel controls are defined when more than 1 hero image', () => {
    render(<Carousel {...carouselListProps} />, {})
    const carouselItems =
      document.getElementsByClassName('carousel-item').length
    if (carouselItems > 1) {
      const prevControl = document.querySelector('.carousel-control-prev')
      const nextControl = document.querySelector('.carousel-control-next')
      const carouselIndicators = document.querySelector('.carousel-indicators')

      expect(prevControl).toBeDefined()
      expect(nextControl).toBeDefined()
      expect(carouselIndicators).toBeInTheDocument()
    }
  })

  test('Check if props are assigned correctly', () => {
    const { container } = render(<Carousel {...carouselListProps} />, {})

    const hero1Img = container.querySelector('.carousel-item:nth-child(1) img')
    const hero1ImgSrc = hero1Img?.getAttribute('data-src')

    const hero1Link = container.querySelector('.carousel-item:nth-child(1) a')
    const hero1LinkHref = hero1Link?.getAttribute('href')

    const hero1H2 = container.querySelector(
      '.carousel-item:nth-child(1) a .container h2'
    )?.textContent

    const hero1H3 = container.querySelector(
      '.carousel-item:nth-child(1) a .container h3'
    )?.textContent
    const hero1Cta = container.querySelector(
      '.carousel-item:nth-child(1) a .container .btn'
    )?.textContent

    expect(hero1ImgSrc).toMatch(
      carouselListProps.heroBannerList[0].background.name
    )
    expect(hero1LinkHref).toMatch(
      carouselListProps.heroBannerList[0].textAndCTA.callToActionURL
    )
    expect(hero1H2).toMatch(
      carouselListProps.heroBannerList[0].textAndCTA.headline
    )
    expect(hero1H3).toMatch(
      carouselListProps.heroBannerList[0].textAndCTA.strapline
    )
    expect(hero1Cta).toMatch(
      carouselListProps.heroBannerList[0].textAndCTA.callToActionText
    )
  })
})

const carouselListProps: CarouselListProps = {
  heroBannerList: [
    {
      _meta: {
        name: 'string',
        schema: 'string',
        deliveryId: '1543512973',
      },
      cssClass: '',
      background: { name: 'homepage-h-d' },
      color: 'white',
      contrastColor: 'dark',
      textAndCTA: {
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
        headline: 'Default headline',
        strapline: 'Default Strapline',
        callToActionText: 'Default call to action',
      },
    },
    {
      _meta: {
        name: 'string',
        schema: 'string',
        deliveryId: '1237612',
      },
      cssClass: '',
      background: { name: 'homepage-h-d' },
      color: 'white',
      contrastColor: 'dark',
      textAndCTA: {
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
        headline: 'Default headline',
        strapline: 'Default Strapline',
        callToActionText: 'Default call to action',
      },
    },
    {
      _meta: {
        name: 'string',
        schema: 'string',
        deliveryId: '983492387',
      },
      cssClass: '',
      background: { name: 'homepage-h-d' },
      color: 'white',
      contrastColor: 'dark',
      textAndCTA: {
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
        headline: 'Default headline',
        strapline: 'Default Strapline',
        callToActionText: 'Default call to action',
      },
    },
  ],
}
