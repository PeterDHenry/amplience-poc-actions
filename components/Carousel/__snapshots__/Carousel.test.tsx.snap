// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component Matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Carousel__StyledCarouselWithIndicators-sc-1y7w2xj-1 lomlWv carousel slide"
    >
      <ol
        class="carousel-indicators"
      >
        <li
          class="active"
        />
        <li />
        <li />
      </ol>
      <div
        class="carousel-inner"
      >
        <div
          class="Carousel__StyledCarouselItem-sc-1y7w2xj-0 jYiBRo active carousel-item"
        >
          <a
            class="d-block position-relative"
            href="https://www.wiggle.co.uk/pro-riding-for-summer"
          >
            <div
              style="position: relative; display: block; padding-bottom: 33.85416666666667%;"
            >
              <noscript />
              <img
                alt="Default hero alt text"
                data-src="https://cdn.media.amplience.net/i/wigglecrcnonprd/CCC-11-5-a?w=1920&h=650&sm=C"
                src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
                style="height: 100%; width: 100%; visibility: hidden; position: absolute; top: 0px; left: 0px;"
              />
              <div
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; background: white; transition: 0.1s opacity;"
              />
            </div>
            <div
              class="container"
            >
              <div
                class="position-relative"
              >
                <div
                  class="position-absolute bottom-0 pb-5 mb-5 text-uppercase"
                >
                  <h2
                    class=""
                  >
                    Default ride like a pro
                  </h2>
                  <div
                    class="btn"
                  >
                    Default shop now
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="Carousel__StyledCarouselItem-sc-1y7w2xj-0 jYiBRo carousel-item"
        >
          <a
            class="d-block position-relative"
            href="https://www.wiggle.co.uk/womens-collection"
          >
            <div
              style="position: relative; display: block; padding-bottom: 33.85416666666667%;"
            >
              <noscript />
              <img
                alt="Default hero alt text"
                data-src="https://cdn.media.amplience.net/i/wigglecrcnonprd/homepageh1d?w=1920&h=650&sm=C"
                src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
                style="height: 100%; width: 100%; visibility: hidden; position: absolute; top: 0px; left: 0px;"
              />
              <div
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; background: white; transition: 0.1s opacity;"
              />
            </div>
            <div
              class="container"
            >
              <div
                class="position-relative"
              >
                <div
                  class="position-absolute bottom-0 pb-5 mb-5 text-uppercase"
                >
                  <h2
                    class=""
                  >
                    Default Tailored for women
                  </h2>
                  <h3
                    class=""
                  >
                    Your best-ever sports collection
                  </h3>
                  <div
                    class="btn"
                  >
                    Default Shop now
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="Carousel__StyledCarouselItem-sc-1y7w2xj-0 jYiBRo carousel-item"
        >
          <a
            class="d-block position-relative"
            href="https://www.wiggle.co.uk/discover-different"
          >
            <div
              style="position: relative; display: block; padding-bottom: 33.85416666666667%;"
            >
              <noscript />
              <img
                alt="Default hero alt text"
                data-src="https://cdn.media.amplience.net/i/wigglecrcnonprd/homepage-h-d?w=1920&h=650&sm=C"
                src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
                style="height: 100%; width: 100%; visibility: hidden; position: absolute; top: 0px; left: 0px;"
              />
              <div
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; background: white; transition: 0.1s opacity;"
              />
            </div>
            <div
              class="container"
            >
              <div
                class="position-relative"
              >
                <div
                  class="position-absolute bottom-0 pb-5 mb-5 text-uppercase"
                >
                  <h2
                    class=""
                  >
                    Default Discover different
                  </h2>
                  <div
                    class="btn"
                  >
                    Default Find out more
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <a
        class="carousel-control-prev"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
          class="carousel-control-prev-icon"
        />
      </a>
      <a
        class="carousel-control-next"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
          class="carousel-control-next-icon"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
